src/mappings/solution.ts

```diff
@@ -48,7 +48,9 @@ function _createSolution(solutionId: string, trade: Trade, batch: Batch, event:
   let solution = new Solution(solutionId)
 
   // Relation with the batch and the trades
-  solution.batch = batch.id
+  // batchID here is for the next batch
+  // so we subtract 1 (one) to point to the proper batch
+  solution.batch = trade.tradeBatchId.minus(new BigInt(1)).toString()
```

Comment:
I'm thinking that this probably needs to be done before.

The error affects the trade id. 
Look for: `_createTrade`

You'll see there
```js
let batchId = getBatchId(event)
...
trade.tradeBatchId = batchId
```

That is wrong. It's using the "current batch id" for the trade event, when it should be `batchId - 1`

Also add the source reason why we decrement one:
`Solutions are submitted always for the previous batch (not the current one). For this reason, we know the trade id is the current batch minus one.`